### PR TITLE
Converting to Typescript

### PR DESCRIPTION
**Pull request by perej3**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import * as axs from "./axiosAPI";
```
Comment (perej3) [inline src/index.ts]: import { GetApiData } from "./http.service"

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { InquirerQ } from "./inquirer";
+import { ShipModule, Value } from "./interfaces";
+
+getAPI().then(response => InquirerQ(response));
+
+async function getAPI() {
```
Comment (perej3) [inline src/index.ts]: rename to `getSharhips()` please

File: src/axiosAPI.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
+
+export async function GetAPIData(url: string) {
```
Comment (perej3) [inline src/axiosAPI.ts]: rename to `httpGet` please

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  let StarshipUrl: string = `https://swapi.co/api/starships/`;
```
Comment (perej3) [inline src/index.ts]: reduntant `:string`

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  let starShipResponse: ShipModule<Value>[] = [];
+
+  while (StarshipUrl != null) {
+    const starShipsApi = await axs.GetAPIData(StarshipUrl);
+
+    for (let i = 0; i < starShipsApi.data.results.length; i++) {
```
Comment (perej3) [inline src/index.ts]: `const i` should be enough no need to use `let`

File: src/index.ts (hunk 2 above)
Comment (perej3) [reply src/index.ts]: also include return type model

example:

```
async function getStarhips(): MyReturnType
```

File: src/inquirer.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import * as inquirer from "inquirer";
+import * as axs from "./axiosAPI";
```
Comment (perej3) [inline src/inquirer.ts]: fix import as stated in the other file

File: src/inquirer.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  ShipModule,
+  Value,
+  PilotAnswerValue,
+  StarshipAnswer,
+  PilotAnswer
+} from "./interfaces";
+
+export function InquirerQ(starShipResponse: ShipModule<Value>[]) {
+  inquirer.prompt(Questions(starShipResponse)).then(answers => {
+    //console.log("FFF", answers.Starships);
```
Comment (perej3) [inline src/inquirer.ts]: remove comment please :)

File: src/inquirer.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import * as axs from "./axiosAPI";
+import {
+  ShipModule,
+  Value,
+  PilotAnswerValue,
+  StarshipAnswer,
+  PilotAnswer
+} from "./interfaces";
+
+export function InquirerQ(starShipResponse: ShipModule<Value>[]) {
```
Comment (perej3) [inline src/inquirer.ts]: add returntype here too please, also function name should be `getAnswers()`

File: src/inquirer.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+  starShipResponse: ShipModule<Value>[]
+): inquirer.ListQuestion<StarshipAnswer & PilotAnswer>[] {
+  let Ship: inquirer.Question<PilotAnswer> = {
+    type: "list",
+    name: "Starship",
+    message: "Select a Starship",
+    choices: starShipResponse
+  };
+
+  let Plot: inquirer.Question<StarshipAnswer> = {
```
Comment (perej3) [inline src/inquirer.ts]: i guess typo it should be `Pilot`

File: src/inquirer.ts [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    type: "list",
+    name: "Pilot",
+    message: "Select Pilot",
+    choices(answers) {
+      // console.log(answers);
```
Comment (perej3) [inline src/inquirer.ts]: remove comment please 👍

File: src/inquirer.ts [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+      return GetPilots(answers.Starship.url);
+    }
+  };
+  return [Ship, Plot];
+}
+
+async function GetPilots(StarshipKey: string[]) {
+  const PilotUrls = StarshipKey;
```
Comment (perej3) [inline src/inquirer.ts]: this const is reduntant

File: src/inquirer.ts [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    message: "Select Pilot",
+    choices(answers) {
+      // console.log(answers);
+      return GetPilots(answers.Starship.url);
+    }
+  };
+  return [Ship, Plot];
+}
+
+async function GetPilots(StarshipKey: string[]) {
```
Comment (perej3) [inline src/inquirer.ts]: add return type please ;)

File: src/interfaces.ts [hunk 13]
```diff
@@ -0,0 +1,29 @@
+export interface ShipModule<T> {
```
Comment (perej3) [inline src/interfaces.ts]: file name should be more specific

example:
```
startships.model.ts
pilot.model.ts
```

File: src/index.ts (hunk 5 above)
Comment (Liam-CurmideGray) [reply src/index.ts]: cannot set const i as it requires to be incremented, therefore was left as let.

**Pull request merged**